ledger/trie.go

```diff
@@ -119,9 +119,9 @@ func (p Path) Equals(o Path) bool {
 
 // DeepCopy returns a deep copy of the payload
 func (p *Path) DeepCopy() Path {
-	newP := make([]byte, len(*p))
-	copy(newP, *p)
-	return Path(newP)
+	newP := make([]byte, 0, len(*p))
+	path := []byte(*p)
+	return Path(append(newP, path...))
```

Comment:
how fast is the alternative version? another option we were thinking previously was to change the path type to include an internal private data and be read only. or make sure cadence is read safe and get rid of this deep copy

Reply:
The difference between the 2 versions is the slice initialization. The older version initializes the new slice to zeros, the new version doesn't. 

Yes, good idea, I guess that also applies to payloads deep copy. 
For paths, another option is to switch the type `Path` to `[32]byte` and then pass parameters by value.